gatsby-node.js

```diff
@@ -0,0 +1,164 @@
+  };
+
+  return knackPagePromise.then(() => {
+    const pageNodes = getNodesByType('File');
+    const knackPageContent = pageNodes
+      .filter(node => node.relativePath === 'rulebook/knacks.mdx')
+      .pop().internal.content;
+
+    const knackAttributes = [
+      ['**Effect:** ', 'effect'],
+      ['**Level:** ', 'level'],
+      ['**Note:** ', 'note'],
+    ];
+    let currentCategory = null;
+    let currentKnack = null;
+    let currentKnackContent = {};
+
+    knackPageContent.split('\n').map(line => {
+      if (line.startsWith('## ')) {
+        currentCategory = line.replace(/^[\s#]+/g, '');
+      } else if (line.startsWith('### ')) {
+        if (currentKnack) {
+          // Write out the current knack and reset state
+          createKnackNode(currentKnackContent);
+          currentKnackContent = {};
+        }
+
+        currentKnack = line.replace(/^[\s#]+/g, '');
+        currentKnackContent['name'] = currentKnack;
+        currentKnackContent['category'] = currentCategory;
+        currentKnackContent['content'] = '';
+      } else if (currentKnack) {
+        currentKnackContent['content'] += line;
+        currentKnackContent['content'] += '\n';
+      }
+
+      knackAttributes.map(([pattern, key]) => {
+        if (line.startsWith(pattern)) {
+          currentKnackContent[key] = line.replace(pattern, '');
+        }
+      });
+    });
+
+    // Write out the last knack
+    createKnackNode(currentKnackContent);
+  });
+};
+
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes, createFieldExtension } = actions;
+
+  createFieldExtension({
+    name: `defaultZero`,
+    extend() {
+      return {
+        resolve(source, args, context, info) {
+          if (source[info.fieldName] == null) {
+            return 0;
+          }
+          return source[info.fieldName];
+        },
+      };
+    },
+  });
+  createFieldExtension({
+    name: `defaultOne`,
+    extend() {
+      return {
+        resolve(source, args, context, info) {
+          if (source[info.fieldName] == null) {
+            return 1;
+          }
+          return source[info.fieldName];
+        },
+      };
+    },
+  });
+
+  const typeDefs = `
+    type Knack implements Node @dontInfer {
+      id: String!
+      category: String!
+      content: String!
+      effect: String!
+      level: String!
+      note: String
+    }
+    type Mdx implements Node @dontInfer {
+      frontmatter: Frontmatter
+    }
+    type Frontmatter {
+      title: String!
+      icon: String!
+      menu_order: Int @defaultZero
+      sheet: CharacterSheet
+    }
+    type CharacterSheet {
+      name: String!
+      player: String!
+      species: Species
+      hook: String!
+      level: Int!
+      xp: Int!
+      attributes: CharacterAttributes!
+      max_hp: Int!
+      hp: Int!
+      coins: Int!
+      knacks: [CharacterKnack]!
+      items: [String]
+    }
+    type CharacterKnack {
+      knack: Knack! @link
```

Comment:
This link seems to work in the graphQL, but yet the additional information doesn't seem to be available in the component context. This has been vexing for me.

This is what it looks like in GraphQL:
<img width="930" alt="Screen Shot 2020-11-26 at 4 44 45 PM" src="https://user-images.githubusercontent.com/22176452/100399211-b3d5c380-3006-11eb-95ef-49a3ff973e60.png">

But in the Component... characterKnack.knack is NOT this object -it is instead just the string as it appears in the `sheet` on `glenn.mdx`. Which surprised/confused me.